common/evmcompat/helpers.go

```diff
@@ -156,15 +156,19 @@ func recoverAddressFromEos(hash []byte, sig []byte) (common.Address, error) {
 
 func recoverAddressFromEosScatter(hash []byte, sig []byte) (common.Address, error) {
 	var signer common.Address
-	if len(sig) != 67 {
-		return signer, fmt.Errorf("eos signature must be 67 bytes, not %d bytes", len(sig))
+
+	if len(sig) != 107 {
+		return signer, fmt.Errorf("eos scatter signature must be 107 bytes, not %d bytes", len(sig))
 	}
-	signature := ecc.NewSigNil()
-	_, err := signature.Unpack(sig[1:])
+	signature, err := ecc.NewSignature(string(sig[6:]))
```

Comment:
The `SignatureType_EOS_SCATTER` byte is still included so should'nt this be 
`signature, err := ecc.NewSignature(string(sig[7:]))`?